packages/server/lib/cloud/protocol.ts

```diff
@@ -125,6 +125,14 @@ class ProtocolManagerImpl implements ProtocolManager {
 
     this.protocol?.commandLogChanged(log)
   }
+
+  viewportChanged (input: any): void {
+    if (!this.protocolEnabled()) {
+      return
+    }
+
+    this.protocol?.viewportChanged(input)
```

Comment:
if the type is `any` here, but the type protocol expects to be defined is `ViewportInput`, does this need to do additional type checking or should the corresponding protocol method have any any/unknown type and do additional type checking to ensure the correct form (which is it doing with the zod schema)?